kernel/mempool.c

```diff
@@ -172,14 +172,25 @@ void k_free(void *ptr)
 
 #if (CONFIG_HEAP_MEM_POOL_SIZE > 0)
 
+#if (CONFIG_HEAP_MEM_POOL_SIZE < CONFIG_HEAP_MEM_POOL_MIN_SIZE)
+#error "The smallest object must have a lower size then the entire heap"
+#endif
+#if (CONFIG_HEAP_MEM_POOL_MIN_SIZE & (CONFIG_HEAP_MEM_POOL_MIN_SIZE - 1)) != 0
+#error "The smallest object size must be a power of two"
```

Comment:
It doesn't, actually.  The mempool code will align and pad as needed so that a block is an integer number of machine words (with a minimum of two words to store the list node).  There is some waste when heap sublevel sizes are not evenly divisible by four, but there's no such requirement on the lowest level -- if you ask for e.g. 12 byte minimum blocks, you can have them without waste.

Reply:
Hi @andyross , thanks. I removed all the checks.